pinot-core/src/test/java/org/apache/pinot/queries/BaseQueriesTest.java

```diff
@@ -91,7 +95,9 @@ protected <T extends Operator> T getOperatorWithFilter(String query) {
   /**
    * Run query on multiple index segments.
    * <p>Use this to test the whole flow from server to broker.
-   * <p>The result should be equivalent to querying 4 identical index segments.
+   * <p>Unless explicitly override getDistinctInstances or initialize 2 distinct index segments in test, the result
```

Comment:
From your code it seems overriding `getDistinctInstances` and initializing 2 instances is the way to trigger the new `getBrokerResponseDistinctInstances ` behavior? Also the user should take care of assigning the distinct segments correctly to the two instances. Maybe update this comment a little bit and also add a pointer to the usage `CovarianceQueriesTest`?